taskweaver/ext_role/document_retriever/document_retriever.py

```diff
@@ -2,29 +2,60 @@
 import os
 import pickle
 
-from taskweaver.plugin import Plugin, register_plugin
+from injector import inject
 
+from taskweaver.logging import TelemetryLogger
+from taskweaver.memory import Memory, Post
+from taskweaver.module.event_emitter import SessionEventEmitter
+from taskweaver.module.prompt_util import PromptUtil
+from taskweaver.module.tracing import Tracing
+from taskweaver.role import Role
+from taskweaver.role.role import RoleConfig, RoleEntry
 
-@register_plugin
-class DocumentRetriever(Plugin):
-    vectorstore = None
 
-    def _init(self):
-        try:
-            import tiktoken
-            from langchain_community.embeddings import HuggingFaceEmbeddings
-            from langchain_community.vectorstores import FAISS
-        except ImportError:
-            raise ImportError("Please install langchain-community first.")
+class DocumentRetrieverConfig(RoleConfig):
+    def _configure(self):
+        # default is the directory where this file is located
+        self.index_folder = self._get_str(
+            "index_folder",
+            os.path.join(
+                os.path.dirname(__file__),
+                "knowledge_base",
+            ),
+        )
+        self.size = self._get_int("size", 5)
+        self.target_length = self._get_int("target_length", 256)
+
+
+class DocumentRetriever(Role):
+    @inject
+    def __init__(
+        self,
+        config: DocumentRetrieverConfig,
+        logger: TelemetryLogger,
+        tracing: Tracing,
+        event_emitter: SessionEventEmitter,
+        role_entry: RoleEntry,
+    ):
+        super().__init__(config, logger, tracing, event_emitter, role_entry)
+        self.enc = None
+        self.chunk_id_to_index = None
+        self.vectorstore = None
+        self.embeddings = None
+
+    def initialize(self):
+        import tiktoken
+        from langchain_community.embeddings import HuggingFaceEmbeddings
+        from langchain_community.vectorstores import FAISS
 
         self.embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
         self.vectorstore = FAISS.load_local(
```

Comment:
add a check if the local indexing file does not exist? remind the users to run the document_indexer.py file?

Reply:
The user needs to configure the path explicitly. So, she needs to make sure the existence already.